### PR TITLE
Improvements in IGMP layer

### PR DESCRIPTION
**Pull request by gx740**
The same changes as in PR #381

### CODE REVIEW

File: Packet++/header/IgmpLayer.h [hunk 1]
```diff
@@ -326,13 +330,13 @@ class IgmpV3QueryLayer : public IgmpLayer
 
 	/**
 	 * A constructor that allocates a new IGMPv3 membership query
-	 * @param[in] multicastAddr The multicast address to set. This is an optional parameter and has a default value of IPv4Address#Zero
+	 * @param[in] multicastAddr The multicast address to set. This is an optional parameter and has a default value of unspecified IPv4 address
```
Comment (seladb) [inline Packet++/header/IgmpLayer.h:333]: Can you add `unspecified/zero` instead of just `unspecified`?

File: Packet++/header/IgmpLayer.h [hunk 2]
```diff
@@ -283,15 +285,17 @@ class IgmpV2Layer : public IgmpLayer
...
 	 */
-	IgmpV2Layer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+	IgmpV2Layer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		: IgmpLayer(data, dataLen, prevLayer, packet, IGMPv2) {}
 
 	/**
 	 * A constructor that allocates a new IGMPv2 header
 	 * @param[in] type The message type to set
-	 * @param[in] groupAddr The multicast address to set. This is an optional parameter and has a default value of IPv4Address#Zero
+	 * @param[in] groupAddr The multicast address to set. This is an optional parameter and has a default value of unspecified IPv4 address
```
Comment (seladb) [inline Packet++/header/IgmpLayer.h:294]: Can you add `unspecified/zero` instead of just `unspecified`?

File: Packet++/header/IgmpLayer.h (hunk 1 above)
Comment (gx740) [reply Packet++/header/IgmpLayer.h]: Done

File: Packet++/header/IgmpLayer.h (hunk 2 above)
Comment (gx740) [reply Packet++/header/IgmpLayer.h]: Done

**Pull request merged**